lib/tag-field.jsx

```diff
@@ -194,6 +196,7 @@ const mapStateToProps = ( {
 	const note = getNote( state );
 	const revision = selectedRevision || note;
 	return {
+		allTags: state.tags.map( property( 'data.name' ) ),
```

Comment:
Yeah, this last commit was to solve some conflicts and keep them in check with how the rest of the PR was written.
It didn't make sense to me to have only the `allTags` prop to be passed all the way from `app.jsx` while everything else is directly mapped from the state.

Once (or "if"? I recall some talk of closing it?) #402 is completed and merged, I'll definitely make sure to update this to reflect your changes.